posts/2023-07-15-riviera-dev.adoc

```diff
@@ -0,0 +1,162 @@
+
+We've all seen layout shifts when surfing the web.
+This can take different forms, such as an image that loads and moves the whole web page at once.
+Or a font that also takes a long time to load and moves all the text once it has been applied.
+In general, these problems occur with DOM elements added dynamically or resources loaded asynchronously.
+
+These Layout Shifts are problematic for users and also for SEO with the Cumulative layout shift (CLS) metric.
+For example, in the following video you can see just how frustrating the problem can be when you do the opposite action to the one initially planned.
+
+.Layout shift example
+image::layoutShift.gif[Layout shift example, width = 800]
+
+The causes of layout shifts fall into four broad categories:
+
+- page layout
+- animations
+- media
+- fonts
+
+Layout shifts are often caused by automatic values that depend on the content.
+In fact, as we can see from the example below taken from https://speakerdeck.com/goetter/tu-fais-du-web-et-tu-ne-connais-pas-les-layout-shifts-nan-mais-dot-dot-dot?slide=20[the slides], a grid container with an auto-flow on the columns is problematic.
+
+.Layout shift https://speakerdeck.com/goetter/tu-fais-du-web-et-tu-ne-connais-pas-les-layout-shifts-nan-mais-dot-dot-dot?slide=20[example]
+image::layout.png[Layout shift example, width = 800]
+
+This CSS code will tell the grid container to generate columns of equivalent size.
+As a result, at the start we only have a single column.
+Then the second column is loaded, so the container splits into two columns, and again the same thing happens when the last column is loaded.
+So there are 4 layout shifts in this little example.
+
+These layout shifts are difficult to detect manually, but fortunately a number of tools exist to make the task easier!
+For example https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?hl=fr[Lightouse] will give you an indication of CLS as well as advice on how to avoid them.
+The Chrome DevTools are also very useful, as they allow you to replay the loading of your page step by step in order to identify layout shifts and detect problematic elements.
+
+If you're interested in this subject, I highly recommend Raphaël's https://speakerdeck.com/goetter/tu-fais-du-web-et-tu-ne-connais-pas-les-layout-shifts-nan-mais-dot-dot-dot[slides], which are packed with examples and advice.
+
+=== Les dessous des noms de domaines (Translation: Behind domain names) - https://rivieradev.fr/orateur/1086[Thomas Fabre]
+This talk was about domain names and how they work.
+Thomas worked for several years in a TLD (Top-Level Domain), AFNIC, which manages the `.fr` domain for example.
+
+Without domain names, our experience of the Internet would be radically different.
+Instead of typing the names of your favourite websites into your browser, you would have to remember the IP addresses of all the servers - a real headache.
```

Comment:
great use of the em dash here! Just a reminder that with em dashes, they should not have a space between the words and the line should be a bit longer. Ex: "servers--a real headache."

Reply:
Thanks, I didn't know that 🙏